compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirDataClassPrimaryConstructorChecker.kt

```diff
@@ -7,23 +7,38 @@ package org.jetbrains.kotlin.fir.analysis.checkers.declaration
 
 import org.jetbrains.kotlin.descriptors.ClassKind
 import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.checkers.hasValOrVar
 import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
 import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
 import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
 import org.jetbrains.kotlin.fir.declarations.FirConstructor
 import org.jetbrains.kotlin.fir.declarations.FirRegularClass
 import org.jetbrains.kotlin.fir.declarations.isData
 
-object FirPrimaryConstructorRequiredForDataClassChecker : FirRegularClassChecker() {
+object FirDataClassPrimaryConstructorChecker : FirRegularClassChecker() {
     override fun check(declaration: FirRegularClass, context: CheckerContext, reporter: DiagnosticReporter) {
         if (declaration.classKind != ClassKind.CLASS || !declaration.isData) {
             return
         }
 
-        val hasPrimaryConstructor = declaration.declarations.any { it is FirConstructor && it.isPrimary }
+        val primaryConstructor = declaration.declarations.firstOrNull { it is FirConstructor && it.isPrimary } as FirConstructor?
```

Comment:
Just `declaration.getPrimaryConstructorIfAny()`?

Reply:
FYI, I'm trying to rename it (https://github.com/JetBrains/kotlin/pull/4142/commits/28bc241fb39ea47234e9305445114c7abb951ea7) to be consistent with all the other utils I'm adding there. Depending on which one goes first, other will need to rebase/update; probably I need to do that, but to not forget that. :)

Reply:
FYI, it's in: https://github.com/JetBrains/kotlin/commit/d348e5818358eda64aca6c6f0f5acaa4b9013e16 Please update this use here.